bindings/cpp/src/ethereum.rs

```diff
@@ -0,0 +1,118 @@
+use anyhow::{anyhow, Result};
+use defi_wallet_core_common::contract::ContractCall;
+use defi_wallet_core_common::contract::DynamicContract;
+use defi_wallet_core_common::EthAbiTokenBind;
+use defi_wallet_core_common::EthError;
+use ethers::abi::Detokenize;
+use ethers::abi::InvalidOutputType;
+use ethers::abi::Token;
+use ethers::prelude::*;
+use ethers::types::transaction::eip2718::TypedTransaction;
+use std::convert::TryFrom;
+
+pub struct EthContract {
+    abi_json: String,
+    rpcserver: String,
+    contract_address: String,
```

Comment:
instead of these three fields, it could just have `contract: DynamicContract` that will be created in `new_eth_contract`, so that `encode` and `call` can just do `self.contract.(...)` instead of creating the contract over and over?

Reply:
ok, i'll check